malaria24/ona/models.py

```diff
@@ -112,6 +112,10 @@ def compile_digest(cls):
         return digest
 
     def get_digest_email_data(self):
+        min_start_day = 31
+        max_start_day = 1
+        start_week_day = datetime.today().strftime("%d %B %Y")
+        end_week_day = datetime.today().strftime("%d %B %Y")
```

Comment:
I think there might be an issue with this approach.

Because we're only looking at the day, it's possible that we first see "1 February", and mark that as the minimum, but then later on we see "7 January". Because we're only comparing on the day, we won't mark 7 Jan as the minimum date, because 7 > 1.

I think we will need to keep track of the full `datetime` objects to determine the minimum and maximum one that we've seen, and then only format them at the very end when we need them.

We can also use the database to calculate the minimums and maximums, I'll give examples, otherwise `datetime.min` might be of help to you, for setting that initial minimum value.

Reply:
Example for calculating date ranges using the database **for each province**:
```
fac_codes = Facility.objects.filter(province=p).values_list('facility_code', flat=True).distinct()
cases = ReportedCase.objects.filter(facility_code__in=fac_codes, digest__isnull=True).order_by('create_date_time')
start_date = cases.first().create_date_time
end_date = cases.last().create_date_time
```

Reply:
Example for calculating date ranges using the database **across all provinces**:
```
fac_codes = Facility.objects.values_list('facility_code', flat=True).distinct()
cases = ReportedCase.objects.filter(facility_code__in=fac_codes, digest__isnull=True).order_by('create_date_time')
start_date = cases.first().create_date_time
end_date = cases.last().create_date_time
```